src/olympia/addons/models.py

```diff
@@ -1634,7 +1634,7 @@ def addon_user_sync(sender=None, instance=None, **kwargs):
     # Basket doesn't care what role authors have or whether they are listed
     # or not, it just needs to be updated whenever an author is added/removed.
     created_or_deleted = 'created' not in kwargs or kwargs.get('created')
-    if created_or_deleted:
+    if created_or_deleted and instance.addon.status != amo.STATUS_DELETED:
```

Comment:
I assume the case when an add-on is deleted (and so the user isn't seen as an author any longer) is captured elsewhere?

Reply:
Add-on deletion is implemented in the `@Addon.on_change` `watch_changes` function since it's actually a `status` change on `Addon`. It's covered by tests.